### PR TITLE
chore: setup android detox testing

### PR DESCRIPTION
**Pull request by valpinkman**
### 📝 Description

Same as iOS, setup Detox testing for Android (using github large runners with hardware acceleration)

### ❓ Context

- **Impacted projects**: `ci`, `llm` 
- **Linked resource(s)**: https://ledgerhq.atlassian.net/browse/LIVE-8586 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

should now run the Android Detox tests

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/package.json [hunk 1]
```diff
@@ -7,10 +7,9 @@
     "pod": "cd ios && bundle exec pod install --repo-update",
     "start": "react-native start",
     "ios": "react-native run-ios",
-    "android": "react-native run-android --appIdSuffix=debug",
+    "android": "react-native run-android --appIdSuffix=debug --active-arch-only",
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-mobile/package.json:10]: Very cool, didn't know about this

File: apps/ledger-live-mobile/detox.config.js [hunk 2]
```diff
@@ -70,6 +70,8 @@ module.exports = {
       device: {
         avdName: "Pixel_5_API_31",
       },
+      gpuMode: "host",
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-mobile/detox.config.js:73]: What's this doing? Maybe worth a comment to explain

Comment (valpinkman) [reply apps/ledger-live-mobile/detox.config.js:73]: Enabled hardware acceleration, it's in the doc

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -165,6 +165,7 @@ jobs:
             const fs = require("fs");
             fs.writeFileSync("summary-${{ matrix.os }}.txt", "${{ steps.test-libs.outcome }}", "utf-8");
       - uses: actions/upload-artifact@v3
+        if: always()
```
Comment (ggilchrist-ledger) [inline .github/workflows/test.yml:168]: What happens if there are no artifacts to upload?

File: apps/ledger-live-mobile/detox.config.js [hunk 4]
```diff
@@ -70,6 +70,8 @@ module.exports = {
       device: {
         avdName: "Pixel_5_API_31",
       },
+      gpuMode: "swiftshader_indirect",
+      headless: true,
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-mobile/detox.config.js:74]:
```suggestion
      headless: process.env.CI ? true : false,
```

File: .github/workflows/test-mobile-e2e.yml [hunk 5]
```diff
@@ -116,118 +124,112 @@ jobs:
...
+        id: turborepo-cache-server
+        uses: ./tools/actions/turborepo-s3-cache
+        with:
+          server-token: "yolo"
+          cleanup-cache-folder: "true"
+          aws-access-key: ${{ secrets.ACCESS_KEY_BUCKET }}
+          aws-secret-key: ${{ secrets.SECRET_KEY_BUCKET }}
+      - name: Install dependencies
+        run: |
+          pnpm i --filter="live-mobile..." --filter="ledger-live" --no-frozen-lockfile --unsafe-perm
```
Comment (ggilchrist-ledger) [inline .github/workflows/test-mobile-e2e.yml:172]:
```suggestion
          pnpm i --filter="live-mobile..." --filter="ledger-live"  --filter="dummy-*" --no-frozen-lockfile --unsafe-perm
```

**Review by ggilchrist-ledger — APPROVED** [review]
Let's do this

**Pull request merged**